executor/write.go

```diff
@@ -931,6 +931,9 @@ func (e *InsertValues) fillRowData(cols []*table.Column, vals []types.Datum, ign
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+	if err = table.CastValues(e.ctx, row, cols, ignoreErr); err != nil {
```

Comment:
why we need `CastValues` before eval GenExpres?

Reply:
Because if the generation expression is `json_extract(a, '$.a')`, but the field `a` is a string, it will fail.